src/compiler/utilities.ts

```diff
@@ -2615,11 +2620,19 @@ namespace ts {
         });
     }
 
-    /** Get the type annotaion for the value parameter. */
-    export function getSetAccessorTypeAnnotationNode(accessor: SetAccessorDeclaration): TypeNode {
+    /** Get the type annotation for the value parameter. */
+    export function getSetAccessorTypeAnnotationNode(accessor: SetAccessorDeclaration, includeJSDocType?: boolean): TypeNode {

```

Comment:
`getSetAccessorTypeAnnotationNode` is used in the transforms as well as the checker. At the point where it is used it is expecting a lexical type annotation node, not an inferred type annotation node.